### PR TITLE
Removed forced check for core.windows.net

### PR DESCRIPTION
**Pull request by AsafMah**
The long term solution is endpoints validation, but this works for now.

### CODE REVIEW

File: kusto/ingest/internal/resources/resources.go [hunk 1]
```diff
@@ -46,7 +46,7 @@ type URI struct {
 // parse parses a string representing a Kutso resource URI.
 func parse(uri string) (*URI, error) {
 	// Regex representing URI that is expected:
-	// https://(\w+).(queue|blob|table).core.windows.net/([\w,-]+)\?(.*)
+	// https://(\w+).(queue|blob|table).[\w.]+/([\w,-]+)\?(.*)
```
Comment (yogilad) [inline kusto/ingest/internal/resources/resources.go:49]: This comment is no longer relevant

File: kusto/ingest/internal/resources/resources.go [hunk 2]
```diff
@@ -57,10 +57,6 @@ func parse(uri string) (*URI, error) {
 		return nil, fmt.Errorf("URI scheme must be 'https', was '%s'", u.Scheme)
 	}
 
-	if !strings.HasSuffix(u.Hostname(), ".core.windows.net") {
-		return nil, fmt.Errorf("URI hostname does not end with '.core.windows.net'")
-	}
-
 	hostSplit := strings.Split(u.Hostname(), ".")
 	if len(hostSplit) != 5 {
```
Comment (yogilad) [inline kusto/ingest/internal/resources/resources.go:61]: the assumption here is _resource.resource_type.core.[dns suffix]_
The condition should better be changed to >4 cause we can't assure the DNS suffix for the cloud would only have two components.

File: kusto/ingest/internal/resources/resources.go [hunk 3]
```diff
@@ -57,10 +57,6 @@ func parse(uri string) (*URI, error) {
...
 		return nil, fmt.Errorf("URI(%s) is invalid: had incorrect URL path before '.core.windows.net'", uri)
```
Comment (yogilad) [inline kusto/ingest/internal/resources/resources.go:62]: This comment is incorrect

```suggestion
		return nil, fmt.Errorf("Storage URI (%s) is invalid'", uri)
```

File: kusto/ingest/internal/resources/resources.go (hunk 1 above)
Comment (AsafMah) [reply kusto/ingest/internal/resources/resources.go:49]: Why? we still check the rest of the uri, I changed the regex to fit any domain.

File: kusto/ingest/internal/resources/resources.go (hunk 2 above)
Comment (AsafMah) [reply kusto/ingest/internal/resources/resources.go]: Fixed

**Review by yogilad — APPROVED** [review]

**Pull request merged**